src/hooks/useCalculator.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from 'react';
+import {
+    caculateResultInput,
+    isValidOperationInput,
+    isValidResultInput,
+} from '../utils';
+
+export const useCalculator = () => {
+    const [digitInput, setDigitInput] = useState<string>('');
+    const [operationInput, setOperationInput] = useState<string>('');
+    const [resultInput, setResultInput] = useState<string>('0');
+    const [operatorPrevNumber, setOperatorPrevNumber] = useState<string>('');
+
+
+    // 1.  들어오는 DigitInput에따라 resultInput 결과를 처리한다 .
+    const handleDigitInput = (e: React.MouseEvent<HTMLButtonElement>) => {
+        
+        // 1.1 숫자 유효성 체크
+        const currentDigitNumber = parseInt(
+            digitInput + e.currentTarget.value
+        ).toString(); //000입력되는거 방지
+
+        if (currentDigitNumber.length > 3) {
+            alert('숫자는 최대 3자리까지 입력가능합니다.');
+            return;
+        }
```

Comment:
```suggestion
        try {
            const currentDigitNumber = getCurrentDigits(e.currentTarget.value);
            // 하위 로직
        } catch (error: unknown) {
            if (error instanceof Error) {
                alert(error.message);
            }
        }
```
에러를 위임해봐도 좋을 것 같아요.
무엇보다 큰 문제를 나눠서 풀어보는 연습을 해보면 좋을 것 같아요.

Reply:
반영하였습니다!